fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -1200,26 +1198,48 @@ public void insertValueBuggy() throws Exception {
 
     @Test
     public void markReadable() throws Exception {
```

Comment:
It seems like it would be good to split this test into multiple tests depending on what state it should be we try to mark it as readable.

Reply:
Would it be ok to break this one test into multiple sub-functions? I.E build data, call test 1, call test 2, etc. 
It seems a waste to build the data separately each time for these mark-index-state tests...

Reply:
Probably the right answer is for there to be test class that statically sets up the records, and then each test can disable and do whatever with the results.
Converting it to a series of sub-functions that indicate what it is doing would be an ok intermediate.
I also don't think this _has_ to be done in this PR.